### ISSUE TITLE
Custom HTML block experience much like Groundhog Day

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#5467]
## Issue Overview
In 2.3.0 the Custom HTML block has been changed to include the recently introduced wp-codeeditor. 
The user experience is not good.  See for yourself. 

## Steps to Reproduce (for bugs)

1. Create a post containing two HTML blocks s below
2. View in the browser, the two blocks look pretty similar
3. Edit Block
4. Warnings appear in the first block regarding indentation
5. Resolve the problems
6. Save
7. Reload the page
8. Go to 4. or move on to the second block
9. Click on Edit as HTML
10. Notice it hasn't really changed.
11. Ignore the annoying warning signs
12. Update
13. Go to 7.
14. Go to 4. or 9.

```
<!-- wp:html -->
<p>Here's a list NOT inside a paragraph</p>
<ul>
    <li>One</li>
    <li>Two</li>
</ul>
<p>which is probably OK HTML</p>
<!-- /wp:html -->

<!-- wp:html -->
<p>Here's a list inside a paragraph
    <ul>
        <li>One</li>
        <li>Two</li>
    </ul> which is probably bad HTML
</p>
<!-- /wp:html -->
```

## Expected Behavior
- Custom HTML blocks that don't prompt me to make unnecessary changes
- Blocks that accept the content as typed without saying it's been changed
- Behaviour similar to the Text widget

## Current Behavior
See the Steps to reproduce and the animated git. 
Additionally, take the following considerations into account
- The HTML | Preview tab should appear whenever you click into a tag as well as text in a different block
- Or when you change cursor from a tag to inner text
- HTML | Preview tab to appear more often when using iPad
- Update button should be enabled whenever the block is changed
- Consistent CSS styling in Preview
- Unexpected console messages
```
 Warning: Failed prop type: You provided a `value` prop to a form field without an `onChange` handler.
```

## Possible Solution
Document the requirements that the Custom HTML block is supposed to satisfy
and develop tests to demonstrate it meets them.

## Screenshots / Video

![issue-groundhog-day](https://user-images.githubusercontent.com/2474435/37089075-4ae81eec-21f7-11e8-90fa-9e925503092a.gif)

## Related Issues and/or PRs

#5123 

## Todos
- [ ] Tests
- [ ] Documentation

**Comment by mtias** [comment]
The HTML block no longer includes syntax highlighting by default.
